src/toil/lib/docker.py

```diff
@@ -1,280 +1,340 @@
-        with attempt:
-            subprocess.check_call(command)
-
-
-def _getContainerName(job):
-    return '--'.join([str(job),
-                      base64.b64encode(os.urandom(9), '-_')]).replace("'", '').replace('_', '')
-
+    try:
+        this_container = client.containers.get(container_name)
+        while this_container.status == 'running':
+            client.containers.get(container_name).stop()
+            this_container = client.containers.get(container_name)
+    except NotFound:
+        _logger.debug("Attempted to stop container, but container != exist: ",
+                      container_name)
+    except requests.exceptions.HTTPError as e:
+        _logger.debug("Attempted to stop container, but server gave an error: ",
+                      container_name)
+        raise create_api_error_from_http_exception(e)
 
 def _containerIsRunning(container_name):
     """
     Checks whether the container is running or not.
+
     :param container_name: Name of the container being checked.
-    :returns: True if running, False if not running, None if the container doesn't exist.
-    :rtype: bool
+    :returns: True if status is 'running', False if status is anything else,
+    and None if the container does not exist.
     """
+    client = docker.from_env(version='auto')
     try:
-        for attempt in retry(predicate=dockerPredicate):
-            with attempt:
-                output = subprocess.check_output(['docker', 'inspect', '--format', '{{.State.Running}}',
-                                                  container_name]).strip()
-    except subprocess.CalledProcessError:
-        # This will be raised if the container didn't exist.
-        _logger.debug("'docker inspect' failed. Assuming container %s doesn't exist.", container_name,
-                   exc_info=True)
+        this_container = client.containers.get(container_name)
+        if this_container.status == 'running':
+            return True
+        elif this_container.status == 'exited':
+            return False
+        elif this_container.status == 'restarting':
+            return False
+        elif this_container.status == 'paused':
+            return False
+    except NotFound:
         return None
-    if output == 'true':
-        return True
-    elif output == 'false':
-        return False
-    else:
-        raise RuntimeError("Got unexpected value for State.Running (%s)" % output)
+    except requests.exceptions.HTTPError as e:
+        _logger.debug("Server error attempting to call container: ",
+                      container_name)
+        raise create_api_error_from_http_exception(e)
+
+def _getContainerName(job):
+    """Create a random string including the job name, and return it."""
+    return '--'.join([str(job),
+                      base64.b64encode(os.urandom(9), '-_')])\
+                      .replace("'", '').replace('"', '').replace('_', '')
+
+def getDeprecatedCmdlineParams(cmdline_params):
```

Comment:
Another possibility is making dockerCall _not_ backwards compatible. If doing so greatly simplifies the command line arguments to dockerCall and makes the code more readable I think it's worth considering.

Reply:
Definitely my first option, I will shoot you an email about it.